pkg/controller/constrainttemplate/constrainttemplate_controller.go

```diff
@@ -144,15 +150,37 @@ func (r *ReconcileConstraintTemplate) Reconcile(request reconcile.Request) (reco
 		return reconcile.Result{}, err
 	}
 
+	dirty := false
+	defer func() {
+		if dirty {
+			r.metrics.registry.report(r.metrics)
+		}
+	}()
+
+	reportError := func() {
```

Comment:
After you move the dirty bit into the registry, you shouldn't need these wrapper/alias functions anymore.